### DISCUSSION TITLE
Scheduler One Task Symfony 6.4

### DISCUSSION THREAD

**Discussion by fumant2000** [symfony/symfony#54216] [Q&A]
### Description

i've developed a symfony 6 application for sms campaigns. lately i needed to integrate a new feature whereby a user could schedule the sending of sms to his product consumers (customer) via the api of an sms provider. lately i've seen that symfony has a new scheduler component allowing to do this, i've already inserted it. but so far i don't understand how to use the bundle and i don't know if it's suitable for my use case because according to my research this bundle only performs recurring tasks. because in my case after sending sms it shouldn't do any more tasks, a bit like symfony's messenger component but this time it will be a scheduled task that will run once. i need help please.

### Example

in my application, the user can decide the exact time he wants the campaign to be launched. for example, if he wants to plan a campaign that will take place on 2024-03-09 at 12h:00, then the task should be planned for that time. and i would like to draw your attention to the fact that this task should only be executed once.

**Comment by NickOver** [comment]
Scheduler is like a cron. You set interval, not exact date.
The only solution i can see is to set time (eg 12:00) and set from and until to date when it should take place.

Imo you should save all that "starts" in db and have cron/scheduler that run command every minut (or every 10, 15 or 30 min) which check is a record that should be started.
Also, you can push messages to queue with infinite retry strategy.

**Comment by curry684** [comment]
I had a really awkward moment just now. I was truly **thrilled** when I read the announcement of the Symfony Scheduler component. Finally, I no longer would have to use [Agenda](https://www.npmjs.com/package/agenda) or Sidekiq or another external solution to do the stuff we are frequently asked to do, but currently have no good way of doing. Today I sat down to look into replacing the `cron` bundled with one of my applications by Scheduler and use all the awesome new features.

I was planning to do all kinds of awesome stuff like this:
```php
$scheduler->in('30 days', new AnonymizeInactiveUser($user->getId()));
// and
$scheduler->in('15 minutes', new ContinueDomainTransferIfCompleted($domainName));
// and
$scheduler->in('2 hours', new CancelTicketReservationIfPaymentNotCompleted($ticketId));
// and
$scheduler->every('15 minutes', new PollWhetherExternalJobHasCompleted($id));
```
...only to discover that none of them are possible, because we have a Scheduler component now that cannot do the one thing its name suggests it would do: schedule jobs. It executes messages at times defined at development time by the developer. It's a cron, not a scheduler.

I'm not saying anyone did a bad job, it's actually a fine cron replacement. It's just mighty confusing that it's called `symfony/scheduler` where it should just be called `symfony/cron` if that's all it can do and all it's planned to do.

But uhm, are there any actual scheduling components around for PHP by now or am I still confined to running a Node container just for Agenda?
